services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,67 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (thermalMgrQueueHandle == NULL)
+    return ERR_CODE_INVALID_STATE;
 
+  if (event == NULL)
+    return ERR_CODE_INVALID_ARG;
+
+  if( xQueueSend(thermalMgrQueueHandle,
+                ( void * ) event,
+                ( TickType_t ) 0) == errQUEUE_FULL)
+  {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t eventCall;
+  eventCall.type = OS_INTERRUPT_EVENT;
+  
+  error_code_t errEvent = thermalMgrSendEvent(&eventCall);
+  if (errEvent != ERR_CODE_SUCCESS)
+    printConsole("Error sending event in OS Handler");
```

Comment:
you should never print directly from an ISR since this function call tries to take a mutex and printing in general isn't reentrant